c7n/cli.py

```diff
@@ -39,21 +39,76 @@ def _default_options(p):
                    help="Only execute named/matched policies")
     p.add_argument("-t", "--resource", default=None, dest='resource_type',
                    help="Only execute policies with the given resource type")
+
     p.add_argument("-v", "--verbose", action="store_true",
                    help="Verbose Logging")
+
+    p.add_argument("--debug", action="store_true",
+                   help="Dev Debug")
+
+    if extended_options:
+        _extended_default_options(p)
+
+
+def _extended_default_options(p):
     p.add_argument(
         "-l", "--log-group", default=None,
         help="Cloudwatch Log Group to send policy logs")
 
-    p.add_argument("--debug", action="store_true",
-                   help="Dev Debug")
     p.add_argument("-s", "--output-dir", required=True,
                    help="Directory or S3 URL For Policy Output")
     p.add_argument("-f", "--cache", default="~/.cache/cloud-custodian.cache")
     p.add_argument("--cache-period", default=60, type=int,
                    help="Cache validity in seconds (Default 60)")
 
 
+def _policy_metrics_options(p):
+    """ Add options specific to policy-metrics subcommand. """
+    _default_options(p, extended_options=False)
+    p.add_argument(
+        '--start', type=date_parse, help='Start date (requires --end, overrides --days)')
+    p.add_argument(
+        '--end', type=date_parse, help='End date')
+    p.add_argument(
+        '--days', type=int, default=14,
+        help='Number of days of history to consider (default: %(default)i)')
+    p.add_argument('--period', type=int, default=60*24*24)
+
+    # The original tools/policymetrics.py set these to None, so I am adding them
+    # as hidden options so they are always None.
+    p.add_argument("--log-group", default=None, help=argparse.SUPPRESS)
```

Comment:
the original policymetrics did need them - it set them to None.  That's why I included them as suppressed options.  The spots they are needed could probably be modified to say something like `if getattr(options, 'log_group')` instead of `if options.log_group` and then we might not need to include them. 

without log_group:

```
$ custodian policy-metrics -c test.yaml 
Traceback (most recent call last):
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/bin/custodian", line 11, in <module>
    load_entry_point('c7n', 'console_scripts', 'custodian')()
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/c7n/cli.py", line 205, in main
    options.command(options)
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/c7n/commands.py", line 47, in _load_policies
    policies = collection.filter(options.policy_filter)
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/c7n/policy.py", line 64, in policies
    policy = Policy(p, self.options)
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/c7n/policy.py", line 400, in __init__
    self.ctx = ExecutionContext(self.session_factory, self, self.options)
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/c7n/ctx.py", line 41, in __init__
    if options.log_group:
AttributeError: 'Namespace' object has no attribute 'log_group'
```

without cache:

```
$ custodian policy-metrics -c test.yaml 
Traceback (most recent call last):
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/bin/custodian", line 11, in <module>
    load_entry_point('c7n', 'console_scripts', 'custodian')()
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/c7n/cli.py", line 205, in main
    options.command(options)
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/c7n/commands.py", line 47, in _load_policies
    policies = collection.filter(options.policy_filter)
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/c7n/policy.py", line 64, in policies
    policy = Policy(p, self.options)
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/c7n/policy.py", line 401, in __init__
    self.resource_manager = self.get_resource_manager()
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/c7n/policy.py", line 477, in get_resource_manager
    return factory(self.ctx, self.data)
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/c7n/resources/ec2.py", line 53, in __init__
    super(EC2, self).__init__(ctx, data)
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/c7n/query.py", line 133, in __init__
    super(QueryResourceManager, self).__init__(data, options)
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/c7n/manager.py", line 38, in __init__
    self._cache = cache.factory(self.ctx.options)
  File "/Users/kkronenfeld/sixfeetup/cloud-custodian/c7n/cache.py", line 31, in factory
    if not config.cache or not config.cache_period:
AttributeError: 'Namespace' object has no attribute 'cache'
```